packages/lesswrong/components/tagging/TagFlagItem.tsx

```diff
@@ -35,38 +36,62 @@ const styles = (theme: ThemeType): JssStyles => ({
   }
 })
 
-const TagFlagItem = ({documentId, showNumber = true, allPages = false, style = "grey", classes }: {
+
+const TagFlagItem = ({documentId, itemType = "tagFlagId", showNumber = true, style = "grey", classes }: {
   documentId?: string,
+  itemType?: "tagFlagId" | "allPages" | "userPages" 
   showNumber?: boolean,
-  allPages?: boolean, 
+  allPages?: boolean,
+  myPages?: boolean,
   style?: "white"|"grey"|"black",
   classes: ClassesType,
 }) => {
   const { LWPopper, ContentItemBody } = Components;
   const {eventHandlers, hover, anchorEl, stopHover } = useHover();
+  const currentUser = useCurrentUser();
   const { document: tagFlag } = useSingle({
     documentId,
     collection: TagFlags,
     fetchPolicy: "cache-first",
     fragmentName: "TagFlagFragment",
   })
-  const multiTerms = allPages ? {view: "allPagesByNewest"} : { view: "tagsByTagFlag", tagFlagId: tagFlag?._id}
+  
+  
+  const getTagFlagItemTerms = (itemType) => { 
```

Comment:
I usually like handling this with an object with keys and values instead of a switch statement. But this is also fine.